### ISSUE TITLE
Rule proposal: no-useless-undefined-return

### ISSUE THREAD

**Issue by AntonioVentilii** [typescript-eslint/typescript-eslint#11686]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [x] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Wherever the code has `return undefined` it will be replaced by `return`

### Fail Cases

```TypeScript
function logSomething(): void {
	return undefined; // ❌ flagged: "Use `return;` instead of `return undefined;`."
}

const doThing = (): void => {
	return undefined; // ❌
};

function maybeNumber() {
	if (Math.random() > 0.5) return undefined; // ❌
}
```

### Pass Cases

```TypeScript
function logSomething(): void {
	return; // ✅ allowed
}

const doThing = (): void => {
	return; // ✅
};

function maybeNumber() {
	if (Math.random() > 0.5) return; // ✅
}
```

### Additional Info

If you want, I can work on it.

**Comment by ronami** [comment]
I think this is already covered in `eslint-plugin-unicorn`'s [`no-useless-undefined`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-useless-undefined.md).

**Comment by bradzacher** [comment]
Yup that rule covers it with a fixer.
You can also trivially do it without a fixer using `no-restricted-syntax`
https://typescript-eslint.io/troubleshooting/faqs/general#how-can-i-ban-specific-language-feature

There's nothing typescript-specific about this request and it is covered completely by known alternatives. As such we shall reject this proposal.
